File: PULSE.Services/Data/Servicings/Servicing.cs
namespace PULSE.Services.Data
{
    public partial class Servicing : AuditableEntity
    {
        public Servicing()
        {
            ServicingParts = new HashSet<ServicingPart>();
        }

        public string? ServicingNumber { get; set; }
        public int? CustomerId { get; set; }
        public int? PaymentId { get; set; }
        public string? CustomerRequest { get; set; }
        public int? Status { get; set; }
        public bool? OnlinePayment { get; set; }
        public decimal? LabourCost { get; set; }
        public string? OfferDetails { get; set; }

        public virtual Customer? Customer { get; set; }
        public virtual Payment? Payment { get; set; }
        public virtual ICollection<ServicingPart> ServicingParts { get; set; }
    }
}

File: pulse_admin/lib/data/servicing/servicing.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:pulse_admin/data/abstract/auditable_model.dart';
import 'package:pulse_admin/data/servicing/servicing_part.dart';
import 'package:pulse_admin/data/servicing/servicing_status.dart';
import 'package:pulse_admin/data/user/customer.dart';

part 'servicing.g.dart';

@JsonSerializable()
class Servicing extends AuditableModel {
  String? servicingNumber;
  int? customerId;
  int? paymentId;
  String? customerRequest;
  int? status;
  String? statusName;
  String? statusNamee;
  bool? onlinePayment;
  double? labourCost;
  String? offerDetails;
  Customer? customer;

  List<ServicingPart> servicingParts = [];

  Servicing();

  factory Servicing.fromJson(Map<String, dynamic> json) =>
      _$ServicingFromJson(json);

  Map<String, dynamic> toJson() => _$ServicingToJson(this);
}
